File: samples/python/cms_sdk_samples/examples/thread_manager.py
#!/usr/bin/env python3
"""
会话管理工具
Thread management tool

Usage:
    python -m cms_sdk_samples.examples.thread_manager list
    python -m cms_sdk_samples.examples.thread_manager get <thread-id>
    python -m cms_sdk_samples.examples.thread_manager messages <thread-id>
    python -m cms_sdk_samples.examples.thread_manager delete <thread-id>
"""

import sys

from ..client import AgentClient, Config, SDKException


def list_threads(client: AgentClient):
    print("📋 会话列表")
    print("-" * 100)

    threads, total = client.list_threads(20)
    print(f"共 {total} 个会话:\n")

    if not threads:
        print("  (无会话)")
        return

    print(f"{'Thread ID':<40} {'标题':<25} {'状态':<10} 创建时间")
    print("-" * 100)

    for t in threads:
        title = t.title
        if title and len(title) > 23:
            title = title[:23] + "..."
        print(f"{t.thread_id:<40} {title or '':<25} {t.status or '':<10} {t.create_time or ''}")


def get_thread(client: AgentClient, thread_id: str):
    print(f"📋 会话详情: {thread_id}")
    print("-" * 60)

    detail = client.get_thread(thread_id)

    print(f"  Thread ID: {detail.thread_id}")
    print(f"  标题: {detail.title}")
    print(f"  状态: {detail.status}")
    print(f"  创建时间: {detail.create_time}")
    print(f"  更新时间: {detail.update_time}")


def list_messages(client: AgentClient, thread_id: str):
    print(f"💬 会话消息: {thread_id}")
    print("-" * 80)

    messages = client.get_thread_data(thread_id, 50)
    print(f"共 {len(messages)} 条消息:\n")

    for i, m in enumerate(messages):
        role_icon = "👤"
        if m.role == "assistant":
            role_icon = "🤖"
        elif m.role == "system":
            role_icon = "⚙️"

        content = m.content
        if content and len(content) > 100:
            content = content[:100] + "..."

        print(f"[{i + 1}] {role_icon} {m.role}")
        if content:
            print(f"    {content}")
        print(f"    时间戳: {m.timestamp}\n")


def delete_thread(client: AgentClient, thread_id: str):
    print(f"🗑️ 删除会话: {thread_id}")

    client.delete_thread(thread_id)
    print("✅ 会话已删除")


def print_usage():
    print("会话管理工具 - CMS SDK 接口演示 (Python)")
    print("=" * 50)
    print("\n用法:")
    print("  python -m cms_sdk_samples.examples.thread_manager <command> [args]")
    print("\n命令:")
    print("  list                  列出所有会话")
    print("  get <thread-id>       获取会话详情")
    print("  messages <thread-id>  列出会话消息")
    print("  delete <thread-id>    删除会话")


def main():
    if len(sys.argv) < 2:
        print_usage()
        sys.exit(1)

    command = sys.argv[1]

    try:
        # Load configuration
        cfg = Config.load_from_env()

        # Create client
        client = AgentClient(cfg)

        if command == "list":
            list_threads(client)
        elif command == "get":
            if len(sys.argv) < 3:
                print("❌ 请指定 thread-id")
                sys.exit(1)
            get_thread(client, sys.argv[2])
        elif command == "messages":
            if len(sys.argv) < 3:
                print("❌ 请指定 thread-id")
                sys.exit(1)
            list_messages(client, sys.argv[2])
        elif command == "delete":
            if len(sys.argv) < 3:
                print("❌ 请指定 thread-id")
                sys.exit(1)
            delete_thread(client, sys.argv[2])
        else:
            print(f"❌ 未知命令: {command}")
            print_usage()
            sys.exit(1)

    except SDKException as e:
        print(f"❌ 错误: {e}")
        sys.exit(1)


if __name__ == "__main__":
    main()

File: samples/typescript/src/examples/thread-manager.ts
#!/usr/bin/env tsx
/**
 * 会话管理工具
 * Thread management tool
 *
 * Usage:
 *   npm run thread-manager -- list
 *   npm run thread-manager -- get <thread-id>
 *   npm run thread-manager -- messages <thread-id>
 *   npm run thread-manager -- delete <thread-id>
 */

import { loadConfigFromEnv, AgentClient, SDKException } from '../client/index.js';

async function listThreads(client: AgentClient): Promise<void> {
  console.log('📋 会话列表');
  console.log('-'.repeat(100));

  const { threads, total } = await client.listThreads(20);
  console.log(`共 ${total} 个会话:\n`);

  if (threads.length === 0) {
    console.log('  (无会话)');
    return;
  }

  console.log(
    `${'Thread ID'.padEnd(40)} ${'标题'.padEnd(25)} ${'状态'.padEnd(10)} 创建时间`
  );
  console.log('-'.repeat(100));

  for (const t of threads) {
    let title = t.title || '';
    if (title.length > 23) {
      title = title.slice(0, 23) + '...';
    }
    console.log(
      `${t.threadId.padEnd(40)} ${title.padEnd(25)} ${(t.status || '').padEnd(10)} ${t.createTime || ''}`
    );
  }
}

async function getThread(client: AgentClient, threadId: string): Promise<void> {
  console.log(`📋 会话详情: ${threadId}`);
  console.log('-'.repeat(60));

  const detail = await client.getThread(threadId);

  console.log(`  Thread ID: ${detail.threadId}`);
  console.log(`  标题: ${detail.title}`);
  console.log(`  状态: ${detail.status}`);
  console.log(`  创建时间: ${detail.createTime}`);
  console.log(`  更新时间: ${detail.updateTime}`);
}

async function listMessages(client: AgentClient, threadId: string): Promise<void> {
  console.log(`💬 会话消息: ${threadId}`);
  console.log('-'.repeat(80));

  const messages = await client.getThreadData(threadId, 50);
  console.log(`共 ${messages.length} 条消息:\n`);

  for (let i = 0; i < messages.length; i++) {
    const m = messages[i];
    let roleIcon = '👤';
    if (m.role === 'assistant') {
      roleIcon = '🤖';
    } else if (m.role === 'system') {
      roleIcon = '⚙️';
    }

    let content = m.content || '';
    if (content.length > 100) {
      content = content.slice(0, 100) + '...';
    }

    console.log(`[${i + 1}] ${roleIcon} ${m.role}`);
    if (content) {
      console.log(`    ${content}`);
    }
    console.log(`    时间戳: ${m.timestamp}\n`);
  }
}

async function deleteThread(client: AgentClient, threadId: string): Promise<void> {
  console.log(`🗑️ 删除会话: ${threadId}`);

  await client.deleteThread(threadId);
  console.log('✅ 会话已删除');
}

function printUsage(): void {
  console.log('会话管理工具 - STAROps SDK 接口演示 (TypeScript)');
  console.log('='.repeat(50));
  console.log('\n用法:');
  console.log('  npm run thread-manager -- <command> [args]');
  console.log('\n命令:');
  console.log('  list                  列出所有会话');
  console.log('  get <thread-id>       获取会话详情');
  console.log('  messages <thread-id>  列出会话消息');
  console.log('  delete <thread-id>    删除会话');
}

async function main() {
  const args = process.argv.slice(2);

  if (args.length < 1) {
    printUsage();
    process.exit(1);
  }

  const command = args[0];

  try {
    // Load configuration
    const cfg = await loadConfigFromEnv();

    // Create client
    const client = new AgentClient(cfg);

    switch (command) {
      case 'list':
        await listThreads(client);
        break;
      case 'get':
        if (args.length < 2) {
          console.log('❌ 请指定 thread-id');
          process.exit(1);
        }
        await getThread(client, args[1]);
        break;
      case 'messages':
        if (args.length < 2) {
          console.log('❌ 请指定 thread-id');
          process.exit(1);
        }
        await listMessages(client, args[1]);
        break;
      case 'delete':
        if (args.length < 2) {
          console.log('❌ 请指定 thread-id');
          process.exit(1);
        }
        await deleteThread(client, args[1]);
        break;
      default:
        console.log(`❌ 未知命令: ${command}`);
        printUsage();
        process.exit(1);
    }
  } catch (e) {
    if (e instanceof SDKException) {
      console.log(`❌ 错误: ${e}`);
    } else {
      console.log(`❌ 错误: ${(e as Error).message}`);
    }
    process.exit(1);
  }
}

main();
